cmd/clusterctl/client/tree/discovery.go

```diff
@@ -151,21 +239,66 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 		}
 	}
 
-	// Handles orphan machines.
-	if len(machineMap) < len(machinesList.Items) {
-		other := VirtualObject(cluster.Namespace, "OtherGroup", "Other")
-		tree.Add(workers, other)
+	return nil
+}
 
-		for i := range machinesList.Items {
-			m := &machinesList.Items[i]
-			if _, ok := machineMap[m.Name]; ok {
-				continue
+func addMachinePoolsToObjectTree(ctx context.Context, c client.Client, cluster *clusterv1.Cluster, workers *unstructured.Unstructured, machinePoolList *expv1.MachinePoolList, tree *ObjectTree, options DiscoverOptions) {
+	for i := range machinePoolList.Items {
+		mp := &machinePoolList.Items[i]
+		addOpts := make([]AddObjectOption, 0)
+		if !options.ShowAllResources {
+			addOpts = append(addOpts, GroupingObject(true))
+		}
+		tree.Add(workers, mp, addOpts...)
+
+		if options.ShowAllResources {
+			if bootstrap, err := external.Get(ctx, c, mp.Spec.Template.Spec.Bootstrap.ConfigRef, cluster.Namespace); err == nil {
+				tree.Add(mp, bootstrap, ObjectMetaName("BootstrapConfig"))
+			}
+
+			if machine, err := external.Get(ctx, c, &mp.Spec.Template.Spec.InfrastructureRef, cluster.Namespace); err == nil {
+				tree.Add(mp, machine, ObjectMetaName("MachineInfrastructure"))
 			}
-			addMachineFunc(other, m)
 		}
+
+		// TODO: Add machinepoolmachines when it's implemented
+	}
+}
+
+func getResourceSetBindingsInCluster(ctx context.Context, c client.Client, namespace string, name string) ([]addonsv1.ClusterResourceSetBinding, error) {
+	if name == "" {
+		return nil, nil
 	}
 
-	return tree, nil
+	resourceSetBindingList := &addonsv1.ClusterResourceSetBindingList{}
+	if err := c.List(ctx, resourceSetBindingList, client.InNamespace(namespace)); err != nil {
```

Comment:
Q: is it required to get all the ClusterResourceSet in the namespace? If I remember well (but might be I'm wrong), there is only one CRSBinding for each Cluster and by convention, it has the same name as the Cluster

Reply:
I don't think we should need to get a list of all the bindings, I just wasn't sure if we could guarantee that there would only be one CRSBinding. If that's the case then we can just fetch one.